### ISSUE TITLE
Feature ask: allow secret reference from KeyVault

### ISSUE THREAD

**Issue by yangl900** [Azure/acs-engine#176]
Today the apimodel contains all the secrets inline, so does the generated ARM template parameter files. ARM natively supports KeyVault reference as parameter.

E.g. my parameter files that I can checkin to repo looks like:

```json
    "apiServerPrivateKey": {
      "reference": {
        "keyVault": {
          "id": "/subscriptions/<sub-id>/resourceGroups/<group>/providers/Microsoft.KeyVault/vaults/<valut>"
        },
        "secretName": "k8s-api-server-private-key"
      }
    }
```

ARM will retrieve the secret safely on deployment time, as user I don't need touch the secrets at all.

It will be nice if acs-engine can:
1) on generating secrets, support pushing the secrets directly to a KeyVault. This leaves the local machine clean.
2) support referencing keyvault from apimodel.json, and generate the same reference in ARM template parameters file.

**Comment by anhowe** [comment]
@yangl900 does @JackQuincy checkin solve this?  He provides examples as well: #135

**Comment by yangl900** [comment]
Unfortunately it doesn't. I'm aware of that change, however that only allows we pass in certificate secret to the VM via Linux profile. It does not resolve secrets like SPN client secret from KV. I think using ARM to resolve will be a more generic approach.

**Comment by anhowe** [comment]
fixed with #223
